errors/error.go

```diff
@@ -43,3 +45,41 @@ type HTTPError interface {
 	Text() string
 	Code() int
 }
+
+type HTTPClientError struct {
+	code    int
+	text    string
+	rawResp string
+}
+
+func NewHTTPClientError(code int, body []byte) error {
```

Comment:
Would moving `HTTPClientError` and `NewHTTPClientError` to `internal/viridian/errors.go`  make sense?